discovery-infra/node_controlers/qe_vm_controler.py

```diff
@@ -0,0 +1,90 @@
+import os
+import re
+import libvirt
+import utils
+
+from  node_controlers.node_controler import NodeControler
+
+class QeVmController(NodeControler):
+
+    def __init__(self):
+        self.libvirt_connection = libvirt.open('qemu:///system')
+
+    def __del__(self):
+        self.libvirt_connection.close()
+
+    def list_nodes(self):
+        nodes = {}
+        domains = self.libvirt_connection.listAllDomains()
+
+        for domain in domains:
+            domain_name = domain.name()
+            if ("master" in domain_name) or ("worker" in domain_name):
+                domain_state = "running" if domain.isActive() else "shut_off"
+                nodes[domain_name] = domain_state
+
+        return nodes
+    
+    def shutdown_node(self, node_name):
+        node = self.libvirt_connection.lookupByName(node_name)
+
+        node.destroy()
+
+        return 
+
+    def shutdown_all_nodes(self):
+        nodes = self.list_nodes()
+
+        for node in nodes.keys():
+            self.shutdown_node(node)
+
+        return
+
+    def start_node(self, node_name):
+        node = self.libvirt_connection.lookupByName(node_name)
+
+        node.create()
+
+        return 
+
+    def start_all_nodes(self):
+        nodes = self.list_nodes()
+
+        for node in nodes.keys():
+            self.start_node(node)
+
+        return
+
+    def restart_node(self, node_name):
+        node = self.libvirt_connection.lookupByName(node_name)
+
+        node.destroy()
```

Comment:
Can `node.destroy()` fail? Maybe try-catch?

Reply:
maybe there is even another option? as for example to kill qemu process?

Reply:
@YuviGold node.destroy() can fail and raise an exception which will exit the test. I think this should would this way, wdty?
@tsorya this seems to aggressive to me. The libvirt python does this through the libvirt api

Reply:
i know but it will really painful to fail the test cause for some reason libvirt failed to stop vm.

Reply:
but ok for the start, will handle those cases one day

Reply:
Added a check to to see that the node is running before shutdown and vice-versa